### ISSUE TITLE
license format documentation

### ISSUE THREAD

**Issue by vincentvanhees** [citation-file-format/cffconvert#69]
I use license LGPL (>= 2.0, < 3), but cffconvert complains that it is not in the list of licenses without providing info on where the list is. I then found out that [this](https://spdx.org/licenses/) is the list, which tells me to use LGPL-2.0-or-later.

Would it be an idea to update line 209 of [citation.py](https://github.com/citation-file-format/cff-converter-python/blob/f1a932b80a5261794012239b9280e046dfb7d87b/cffconvert/citation.py
) from:
`raise ValueError("Provided license {0} not in list of licenses".format(spdx_license_code))`

to:
`raise ValueError("Provided license {0} not in list of licenses, see https://spdx.org/licenses/ for possible licenses".format(spdx_license_code))`

**Comment by vincentvanhees** [comment]
or simply:
`raise ValueError("Provided license {0} not in list of licenses: https://spdx.org/licenses/".format(spdx_license_code))`

**Comment by jspaaks** [comment]
I think that's a good idea, but we should find a way to differentiate between the different versions of spdx that can be used in different versions of cff itself. e.g.

cff-version 1.0.3:
https://github.com/citation-file-format/schema/blob/a9d6b8eee4642da18564f09353ae8c60b831b2c4/CFF-Core/schema.yaml#L68-L73

**Comment by vincentvanhees** [comment]
in that case it may be easier to have a separate wikipage or readmefile where the license formatting options are explained?

**Comment by jspaaks** [comment]
Myeah or if you get this error while validating, return the part of the schema that has the list of allowable entries (with some formatting, because currently it's difficult to read).

**Comment by jspaaks** [comment]
Filed issue https://github.com/citation-file-format/citation-file-format/issues/59

**Comment by c-martinez** [comment]
cffconvert V2.0.0 already mentions a list of possible options (although the list is quite long...): 

```shell
jsonschema.exceptions.ValidationError: 'Apache dos' is not one of ['0BSD', 'AAL', 'Abstyles', 'Adobe-2006', ...
```

Probably error messages could be more user friendly, but that is a different issue. Can this issue be closed then?

**Comment by jspaaks** [comment]
I've filed issue https://github.com/citation-file-format/citation-file-format/issues/450 as well, once that's fixed it will show up in the output that `cffconvert` generates. Together with the recently merged "truncated output" feature (PR #290), I think that should be enough to figure it out, although I admit it could be a bit more user friendly.

Closing this issue.
